### DISCUSSION TITLE
Which Java versions do we want to cover?

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#910] [General]
Inspired by #740 I would like to discuss which Java versions we want to officially support.
These versions should be clearly communicated by us (e.g., in the README file) and also covered by our tests (see #841).
How do you like it if we focus only on the long-term support (LTS) versions of Java (currently: `8`, `11`, `17`)?
This is the way Apache Spark handles it, afaik.
Are there any other ideas/preferences?

**Comment by dominikriemer** [comment]
I'd prefer to switch to Java 17 after the next release so that we can use the latest Spring & Spring Boot dependencies which require Java 17.
Officially supporting only LTS versions sounds good to me!

**Reply by bossenti** [reply]
So in this case we would only support Java 17 due to the missing downwards compatibility of spring, right?
Not an issue in my point of view, just to make it clear

**Reply by bossenti** [reply]
See https://github.com/apache/streampipes/pull/1252 adding a proper statement to the README
